File: src/main/java/application/history/history/application/services/CreateHistoryService.java
package application.history.history.application.services;

import application.history.history.application.inputs.CreateHistoryInput;
import application.history.history.domain.entities.History;
import application.history.history.domain.repositories.HistoryRepository;
import org.springframework.stereotype.Service;

import java.util.UUID;

@Service
public class CreateHistoryService {
    private final HistoryRepository repository;

    public CreateHistoryService(HistoryRepository repository) {
        this.repository = repository;
    }

    public History run(CreateHistoryInput request) {
        return this.repository.save(new History(
            UUID.randomUUID().toString(), request.getContent(), request.getDate()
        ));
    }
}

File: src/history/main/kotlin/com/example/history/application/services/CreateHistoryService.kt
package com.example.history.application.services

import com.example.history.application.inputs.CreateHistoryInput
import com.example.history.domain.entities.History
import com.example.history.domain.repositories.HistoryRepository
import org.springframework.stereotype.Service
import java.util.*

@Service
class CreateHistoryService(repository: HistoryRepository) {
    private val repository: HistoryRepository
    fun run(request: CreateHistoryInput): History {
        return repository.save(
            History(
                UUID.randomUUID().toString(), request.getContent(), request.getDate()
            )
        )
    }

    init {
        this.repository = repository
    }
}
